packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/modeledDataStreamTable.tsx

```diff
@@ -98,19 +98,30 @@ export function ModeledDataStreamTable({
     sorting: {},
   });
 
+  const actionsRef = useRef(actions);
+  const selectedAssetRef = useRef(selectedAsset);
+
+  useEffect(() => {
+    actionsRef.current = actions;
+  }, [actions]);
```

Comment:
Why do you need a ref for the actions? It's not immediately clear to me.

Reply:
Was trying to get a way to not have actions re-render everything. Adding actions as a dependency in the dependency array causes the hook to run on every change because useCollection returns a new action reference.